### PR TITLE
Fix requirements using VAL's Parser

### PR DESCRIPTION
**Pull request by Rowan441**
Used VAL to check the requirements for each `domain.pddl` file and update their requirements based on that

I've checked that there are no parsing errors after the change and got:

```
Error in domain: classical-domains/classical/tyreworld/domain.pddl
Error in domain: classical-domains/classical/logistics00/domain.pddl
Error in domain: classical-domains/classical/logistics00/domain.pddl
Error in domain: classical-domains/classical/data-network-opt18/domain.pddl
Error in domain: classical-domains/classical/data-network-sat18/domain.pddl
Ran on 1056 domain files
```

None of these errors are related to requirements

NOTE: `data-network-opt18` and `data-network-sat18` domains were unchanged because of a [bug in VAL](https://github.com/KCL-Planning/VAL/issues/59)

Also, removed the settlers domain, because it is not a classical domain

### CODE REVIEW

**Comment by Rowan441** [comment]
You can check the files are good for VAL yourself with:

```python
#!/usr/bin/env python3
import os, subprocess
import re
import planning_domains_api as api


domain_ids = map(lambda x: x["domain_id"], api.find_domains(""))
runs = 0
for domain_id in domain_ids:

    domain_paths = set()
    problems = api.get_problems(domain_id)
    for problem in problems:

        # We only need to look at each domain.pddl file once
        domain_path = problem["domain_path"]
        domain_path = os.path.join("classical-domains", *domain_path.split("/")) # Fix domain_path on window machines
        if domain_path in domain_paths:
            continue
        domain_paths.add(domain_path)
        
        runs += 1

        ## Run VAL's Parser on the file:
        val_output = str(subprocess.check_output(["./Parser", domain_path]))

        # Check for errors in output
        if not re.search("Errors: 0, warnings: 0", val_output):
            print(f"Error in domain: {domain_path}")

print(f"Ran on {runs} domain files")
```

File: classical/airport-adl/domain.pddl [hunk 1]
```diff
@@ -9,7 +9,7 @@
 
 
 (define (domain airport)
-(:requirements :adl)
```
Comment (haz) [inline classical/airport-adl/domain.pddl]: This type of change seems like a set-back. Can you detail the protocol to remove `:adl` in lieu of the several smaller ones?

File: classical/psr-large/domain.pddl [hunk 2]
```diff
@@ -1,5 +1,5 @@
 (define (domain psr)
-  (:requirements :adl :derived-predicates)
+  (:requirements :equality :typing :derived-predicates :disjunctive-preconditions :negative-preconditions)
```
Comment (haz) [inline classical/psr-large/domain.pddl:2]: `:derived-predicates` is a heavy hitter. Can you pull up all the domains that happen to have the requirement? We should double check that they do indeed need it.

**Review by haz — CHANGES_REQUESTED** [review]
Just a few questions I want to sort out first...

File: classical/airport-adl/domain.pddl (hunk 1 above)
Comment (Rowan441) [reply classical/airport-adl/domain.pddl]: Was using:
```python
ADL_REQS = {":strips",
    ":typing",
    ":disjunctive-preconditions",
    ":equality",
    ":quantified-preconditions",
    ":conditional-effects"
}
```
`:strips` shouldn't be there I guess, but also `quanitfied-precondition` maybe is an alias?

I got the conditions from https://planning.wiki/ref/pddl/requirements#adl

File: classical/psr-large/domain.pddl (hunk 2 above)
Comment (Rowan441) [reply classical/psr-large/domain.pddl:2]: VAL says:
106: http://www.haz.ca/planning-domains/classical/philosophers/domain.pddl
48: http://www.haz.ca/planning-domains/classical/psr-middle/domain.pddl
41: http://www.haz.ca/planning-domains/classical/psr-large/domain.pddl
92: http://www.haz.ca/planning-domains/classical/optical-telegraphs/domain.pddl

psr-middle and psr-large have `:derived-predicates` in the stated `:requirements` the other two do not

File: classical/airport-adl/domain.pddl (hunk 1 above)
Comment (Rowan441) [reply classical/airport-adl/domain.pddl]: `:quantified-preconditions` doesn't appear in the classical-domains at all btw

**Review by haz — APPROVED** [review]

**Comment by haz** [comment]
@Rowan441 Thanks! The race is now on to get the database updated to reflect the more accurate requirements.

**Pull request merged**